### PR TITLE
Traced Sql package fixes (fixes #34)

### LINKED ISSUE #34: Traced Sql package fixes
## Is your feature request related to a problem? Please describe

trace SQL contains the QueryRow method which does not defer the span success.
Various span metadata are wrong or missing due to copy-paste.

## Describe the solution

Add a defer.

### PR DESCRIPTION
**Pull request by mantzas**
## Which problem is this PR solving?

Fixing issue #34.

## Short description of the changes

Fixed a lot of errors during copy paste operations.

### CODE REVIEW

File: service_test.go [hunk 1]
```diff
@@ -43,15 +45,14 @@ func TestServer_Run_Shutdown(t *testing.T) {
...
 	}{
-		{name: "success", cp: &testComponent{}, port: "50000", wantErr: false},
-		{name: "failed to run", cp: &testComponent{errorRunning: true}, port: "50001", wantErr: true},
+		{name: "success", cp: &testComponent{}, wantErr: false},
+		{name: "failed to run", cp: &testComponent{errorRunning: true}, wantErr: true},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			os.Setenv("PATRON_HTTP_DEFAULT_PORT", tt.port)
+			os.Setenv("PATRON_HTTP_DEFAULT_PORT", getRandomPort())
```
Comment (cpapidas) [inline service_test.go:55]: Question: why do we want the `getRandomPort` function and don't set a hard-coded number?

File: sync/http/component_test.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func TestComponent_ListenAndServe_DefaultRoutes_Shutdown(t *testing.T) {
 
 func TestComponent_ListenAndServeTLS_DefaultRoutes_Shutdown(t *testing.T) {
 	rr := []Route{NewRoute("/", "GET", nil, true, nil)}
-	s, err := New(Routes(rr), Secure("testdata/server.pem", "testdata/server.key"), Port(50001))
+	s, err := New(Routes(rr), Secure("testdata/server.pem", "testdata/server.key"), Port(50003))
```
Comment (cpapidas) [inline sync/http/component_test.go:60]: Question: Why do we change the port from 50001 to 50003?

File: service_test.go (hunk 1 above)
Comment (mantzas) [reply service_test.go:55]: The service does not have a mechanism to be stopped.
The error case stops successfully due to an error but the success will stop only if it is killed (SIGTERM etc.)
The test ends but the server does not shut down, and as a result, subsequent tests fail occasionally.
I had 2 options:
- Implement a shutdown feature, only for testing usage
- Randomize the ports so the new test run get's a new port
this is a hack but introducing code only for the test is not something I like either.
Any suggestions?

File: sync/http/component_test.go (hunk 2 above)
Comment (mantzas) [reply sync/http/component_test.go:60]: with the above, this might not be needed...

File: service_test.go (hunk 1 above)
Comment (cpapidas) [reply service_test.go:55]: Theoretically, `getRandomPort`  can return the same port twice we can avoid by check if the port is available.

* The following code adds extra overhead. *

something like:
```go
func getRandomPort() string {
	rnd:= 50000+rand.Int63n(10000)
	port := strconv.FormatInt(rnd, 10)
	ln, err := net.Listen("tcp", ":" + port)
	for  err != nil {
		rnd:= 50000+rand.Int63n(10000)
		port := strconv.FormatInt(rnd, 10)
		ln, err = net.Listen("tcp", ":" + port)
	}
	ln.Close()
	return port
}
``

**Review by cpapidas — APPROVED** [review]

**Pull request merged**